### PR TITLE
Make project name consistent in source, fix licensing issues and update all POs

### PR DESCRIPTION
**Pull request by zenseii**
This changes strings in source to be consistent with the name of the project and to respect licenses and trademark.

All POs have been updated with these new strings.

The translated strings still have to be updated.

None of the names etc and phrasings are final, so please come with input.

The important changes here lie in the source files since all PO files will be updated accordingly to those.

This will need to be decided by the core team so I am tagging you again here.
It would be great if you guys have the time to look at this @ihhub, @oleg-derevenetz and @idshibanov!

### CODE REVIEW

File: files/lang/be.po [hunk 1]
```diff
@@ -3017,7 +3016,7 @@ msgstr ""
 msgid "Original project before 0.7"
 msgstr ""
 
-msgid "Original Heroes of Might and Magic II team"
+msgid "Original Heroes II team"
```
Comment (ihhub) [inline files/lang/be.po:3019]: This is the correct phrase. All rights belong to the original team of Heroes of Might and Magic II game.

File: files/lang/be.po [hunk 2]
```diff
@@ -3199,9 +3197,7 @@ msgstr ""
 msgid "Game Settings"
 msgstr ""
 
-msgid ""
-"Either Roland's or Archibald's campaign from the original Heroes of Might "
-"and Magic II."
+msgid "Either Roland's or Archibald's campaign from the original Heroes II."
```
Comment (ihhub) [inline files/lang/be.po:3200]: This is the correct wording which comes from the original game. Please revert it.

File: src/fheroes2/game/fheroes2.cpp [hunk 3]
```diff
@@ -51,7 +51,7 @@ namespace
 {
     std::string GetCaption()
     {
-        return std::string( "Free Heroes of Might and Magic II, version: " + Settings::GetVersion() );
+        return std::string( "fheroes2, version: " + Settings::GetVersion() );
```
Comment (ihhub) [inline src/fheroes2/game/fheroes2.cpp:54]: Let's call it `fheroes2 engine, version:`

File: src/fheroes2/game/game_credits.cpp [hunk 4]
```diff
@@ -144,7 +144,7 @@ namespace
         output.fill( 0 );
         output._disableTransformLayer();
 
-        Text caption( "Free Heroes of Might and Magic II (" + Settings::GetVersion() + ")", Font::YELLOW_BIG );
+        Text caption( "fheroes2 (" + Settings::GetVersion() + ")", Font::YELLOW_BIG );
```
Comment (ihhub) [inline src/fheroes2/game/game_credits.cpp:147]: `"fheroes2 engine ("`

File: src/fheroes2/game/game_credits.cpp [hunk 5]
```diff
@@ -362,7 +362,7 @@ namespace
         const fheroes2::FontType titleFontType = fheroes2::FontType::normalYellow();
         const fheroes2::FontType nameFontType = fheroes2::FontType::normalWhite();
 
-        fheroes2::Text title( _( "Original Heroes of Might and Magic II team" ), nameFontType );
+        fheroes2::Text title( _( "Original Heroes II team" ), nameFontType );
```
Comment (ihhub) [inline src/fheroes2/game/game_credits.cpp:365]: Please revert this change.

File: src/fheroes2/game/game_credits.cpp [hunk 6]
```diff
@@ -503,7 +503,7 @@ namespace
...
```
Comment (ihhub) [inline src/fheroes2/game/game_credits.cpp:506]: Please revert this change.

File: src/fheroes2/game/game_newgame.cpp [hunk 7]
```diff
@@ -208,7 +208,7 @@ fheroes2::GameMode Game::CampaignSelection()
             return fheroes2::GameMode::MAIN_MENU;
 
         if ( le.MousePressRight( buttonSuccessionWars.area() ) ) {
-            Dialog::Message( _( "Original Campaign" ), _( "Either Roland's or Archibald's campaign from the original Heroes of Might and Magic II." ), Font::BIG );
+            Dialog::Message( _( "Original Campaign" ), _( "Either Roland's or Archibald's campaign from the original Heroes II." ), Font::BIG );
```
Comment (ihhub) [inline src/fheroes2/game/game_newgame.cpp:211]: Please revert this change.

File: src/resources/fheroes2.rc [hunk 8]
```diff
@@ -48,12 +48,12 @@ BEGIN
         BEGIN
             VALUE "Comments",         "Visit us at https://ihhub.github.io/fheroes2\0"
             VALUE "CompanyName",      "fheroes2 Resurrection team\0"
-            VALUE "FileDescription",  "Free implementation of Heroes of Might and Magic II game engine\0"
+            VALUE "FileDescription",  "Free game engine reimplementation of HoMM2 game engine\0"
```
Comment (ihhub) [inline src/resources/fheroes2.rc:51]: Please revert this change. This is legally correct.

File: src/resources/fheroes2.rc [hunk 9]
```diff
@@ -48,12 +48,12 @@ BEGIN
...
             VALUE "FileVersion",      FH2_VERSION_STR
             VALUE "InternalName",     "fheroes2\0"
             VALUE "LegalCopyright",   "\251 2022 fheroes2 Resurrection team <fhomm2@gmail.com>. All rights for the original HoMM II game belong to Ubisoft\0"
             VALUE "OriginalFilename", "fheroes2.exe\0"
-            VALUE "ProductName",      "Free Heroes of Might and Magic II engine\0"
+            VALUE "ProductName",      "fheroes2 Resurrection engine\0"
```
Comment (ihhub) [inline src/resources/fheroes2.rc:56]: `fheroes2 engine` is better.

File: files/lang/pl.po [hunk 10]
```diff
@@ -2441,8 +2441,8 @@ msgstr "Zapisz bieżącą grę."
 msgid "Quit"
 msgstr "Wyjście"
 
-msgid "Quit out of Free Heroes of Might and Magic II."
-msgstr "Zamknij Free Heroes of Might and Magic II."
+msgid "Quit out of fheroes2."
+msgstr ""
```
Comment (zenseii) [inline files/lang/pl.po:2445]:
```suggestion
msgstr "Zamknij fheroes2."
```

File: files/lang/de.po [hunk 11]
```diff
@@ -2635,8 +2635,8 @@ msgstr "Das aktuelle Spiel speichern."
 msgid "Quit"
 msgstr "Beenden"
 
-msgid "Quit out of Free Heroes of Might and Magic II."
-msgstr "Free Heroes of Might and Magic II beenden."
+msgid "Quit out of fheroes2."
+msgstr ""
```
Comment (zenseii) [inline files/lang/de.po:2639]:
```suggestion
msgstr "fheroes2 beenden."
```

File: files/lang/fr.po [hunk 12]
```diff
@@ -2584,8 +2584,8 @@ msgstr "Enregistrer le jeu en cours."
 msgid "Quit"
 msgstr "Quitter"
 
-msgid "Quit out of Free Heroes of Might and Magic II."
-msgstr "Quittez le jeu Free Heroes of Might and Magic II."
+msgid "Quit out of fheroes2."
+msgstr ""
```
Comment (zenseii) [inline files/lang/fr.po:2588]:
```suggestion
msgstr "Quittez fheroes2."
```

File: files/lang/it.po [hunk 13]
```diff
@@ -2432,8 +2432,8 @@ msgstr "Salva la partita in corso."
 msgid "Quit"
 msgstr "Esci"
 
-msgid "Quit out of Free Heroes of Might and Magic II."
-msgstr "Esci da Free Heroes of Might and Magic II."
+msgid "Quit out of fheroes2."
+msgstr ""
```
Comment (zenseii) [inline files/lang/it.po:2436]:
```suggestion
msgstr "Esci da fheroes2."
```

File: src/resources/fheroes2.rc [hunk 14]
```diff
@@ -48,12 +48,12 @@ BEGIN
         BEGIN
             VALUE "Comments",         "Visit us at https://ihhub.github.io/fheroes2\0"
             VALUE "CompanyName",      "fheroes2 Resurrection team\0"
-            VALUE "FileDescription",  "Free implementation of Heroes of Might and Magic II game engine\0"
+            VALUE "FileDescription",  "Free implementation of the Heroes of Might and Magic II game engine\0"
```
Comment (zenseii) [inline src/resources/fheroes2.rc:51]: Maybe it is better to call it this than implementation because implementation sounds like we are actually using the original engine in any way, which is wrong.
```suggestion
            VALUE "FileDescription",  "Free  engine recreation of the Heroes of Might and Magic II game engine\0"
```

Comment (Branikolog) [reply src/resources/fheroes2.rc:51]: Hi, @zenseii !
Is it ok to use `engine` twice in the same caption?

File: files/lang/be.po [hunk 15]
```diff
@@ -2301,7 +2301,7 @@ msgstr ""
 msgid "Quit"
 msgstr ""
 
-msgid "Quit out of Free Heroes of Might and Magic II."
+msgid "Quit out of fheroes2."
```
Comment (ihhub) [inline files/lang/be.po:2304]: I am leaning towards an idea to just have `"Quit out of Heroes of Might and Magic II."`. The reason behind this is that technically users play HoMM2, not the engine itself. Heroes of Might and Magic II game is powered by fheroes2 engine.

File: files/lang/be.po [hunk 16]
```diff
@@ -3178,7 +3177,7 @@ msgid ""
 "Enjoy the game!"
 msgstr ""
 
-msgid "Quit Heroes of Might and Magic and return to the operating system."
+msgid "Quit fheroes2 and return to the operating system."
```
Comment (ihhub) [inline files/lang/be.po:3180]: The same comment as above: `"Quit Heroes of Might and Magic II and return to the operating system."` would be good enough.

File: files/lang/be.po [hunk 17]
```diff
@@ -3149,8 +3149,7 @@ msgid "Greetings!"
 msgstr ""
 
 msgid ""
-"Welcome to Free Heroes of Might and Magic II! Before starting the game "
-"please choose game resolution."
+"Welcome to fheroes2! Before starting the game please choose game resolution."
```
Comment (ihhub) [inline files/lang/be.po:3152]: How about `"Welcome to Heroes of Might and Magic II powered by fheroes2 engine!"`?

File: src/resources/fheroes2.rc (hunk 14 above)
Comment (ihhub) [reply src/resources/fheroes2.rc:51]: `Free implementation of the Heroes of Might and Magic II game engine` sounds legally fine. We state this in the description of the project.

**Review by ihhub — COMMENTED** [review]
Hi @zenseii , I left multiple suggestions in this pull request. Would you mind to take a look?

**Comment by ihhub** [comment]
@zenseii , great thanks for fixing this rights related issue!
